tsl/test/expected/dist_move_chunk.out

```diff
@@ -611,6 +611,162 @@ WHERE hypertable_name = 'dist_test';
  _timescaledb_internal._dist_hyper_3_12_chunk | {db_dist_move_chunk_3}
 (4 rows)
 
+DROP TABLE dist_test;
+-- Ensure dirty chunks are handled during chunk copy
+CREATE TABLE dist_test(time timestamp NOT NULL, device int, temp float);
+SELECT create_distributed_hypertable('dist_test', 'time', 'device', 3);
+WARNING:  column type "timestamp without time zone" used for "time" does not follow best practices
+ create_distributed_hypertable 
+-------------------------------
+ (4,public,dist_test,t)
+(1 row)
+
+INSERT INTO dist_test SELECT t, (abs(timestamp_hash(t::timestamp)) % 10) + 1, 0.10 FROM generate_series('2018-03-02 1:00'::TIMESTAMPTZ, '2018-03-08 1:00', '1 hour') t;
+SELECT * from show_chunks('dist_test');
+                 show_chunks                  
+----------------------------------------------
+ _timescaledb_internal._dist_hyper_4_13_chunk
+ _timescaledb_internal._dist_hyper_4_14_chunk
+ _timescaledb_internal._dist_hyper_4_15_chunk
+ _timescaledb_internal._dist_hyper_4_16_chunk
+(4 rows)
+
+SELECT * FROM test.remote_exec(NULL, $$ SELECT * from show_chunks('dist_test'); $$);
+NOTICE:  [db_dist_move_chunk_1]:  SELECT * from show_chunks('dist_test')
+NOTICE:  [db_dist_move_chunk_1]:
+show_chunks                                 
+--------------------------------------------
+_timescaledb_internal._dist_hyper_4_13_chunk
+_timescaledb_internal._dist_hyper_4_16_chunk
+(2 rows)
+
+
+NOTICE:  [db_dist_move_chunk_2]:  SELECT * from show_chunks('dist_test')
+NOTICE:  [db_dist_move_chunk_2]:
+show_chunks                                 
+--------------------------------------------
+_timescaledb_internal._dist_hyper_4_14_chunk
+(1 row)
+
+
+NOTICE:  [db_dist_move_chunk_3]:  SELECT * from show_chunks('dist_test')
+NOTICE:  [db_dist_move_chunk_3]:
+show_chunks                                 
+--------------------------------------------
+_timescaledb_internal._dist_hyper_4_15_chunk
+(1 row)
+
+
+ remote_exec 
+-------------
+ 
+(1 row)
+
+-- create dirty chunk
+\c :DATA_NODE_2 :ROLE_CLUSTER_SUPERUSER;
+CREATE TABLE _timescaledb_internal._dist_hyper_4_13_chunk(time timestamp NOT NULL, device int, temp float);
+\c :TEST_DBNAME :ROLE_CLUSTER_SUPERUSER;
+-- try to copy chunk copy a chunk with dirty chunk table exists
+CALL timescaledb_experimental.copy_chunk(chunk=>'_timescaledb_internal._dist_hyper_4_13_chunk', source_node=> :'DATA_NODE_1', destination_node => :'DATA_NODE_2');
```

Comment:
Should also test the opposite copy direction:

```
CALL timescaledb_experimental.copy_chunk(chunk=>'_timescaledb_internal._dist_hyper_4_13_chunk', source_node=> :'DATA_NODE_2', destination_node => :'DATA_NODE_1');
```

I guess in this case an error should be raised?

Reply:
Btw, should we introduce an "fail_if_exists" option to the function? I.e., we fail with an error if the chunk exists already unless we set `fail_if_exists=>false`.